ports/atmel-samd/supervisor/qspi_flash.c

```diff
@@ -166,21 +166,34 @@ bool spi_flash_write_data(uint32_t address, uint8_t* data, uint32_t length) {
 bool spi_flash_read_data(uint32_t address, uint8_t* data, uint32_t length) {
     samd_peripherals_disable_and_clear_cache();
 
-    #ifdef EXTERNAL_FLASH_QSPI_DUAL
+    #ifdef EXTERNAL_FLASH_QSPI_SINGLE
+    QSPI->INSTRCTRL.bit.INSTR = CMD_READ_DATA;
+    uint32_t mode = QSPI_INSTRFRAME_WIDTH_SINGLE_BIT_SPI;
+    #elif defined EXTERNAL_FLASH_QSPI_DUAL
```

Comment:
Ideally we'd like to use `#if` here rather than `#ifdef`, and guarantee that all these options are either 1 or 0. This can be cleaned up later, not now, since it affects other ports as well. But could you use parentheses here: `defined(EXTERNAL_FLASH_QSPI_DUAL)`?